src/core/src/node.cpp

```diff
@@ -814,22 +823,25 @@ bool ov::Node::constant_fold(OutputVector& output_values, const OutputVector& in
     if (!all_constants)
         return false;
 
-    HostTensorVector input_tensors;
+    TensorVector input_tensors;
     for (const auto& input : input_values) {
-        auto host_tensor = make_shared<ngraph::runtime::HostTensor>(
-            ov::as_type_ptr<ngraph::op::v0::Constant>(input.get_node_shared_ptr()));
-        input_tensors.push_back(host_tensor);
+        auto constant = ov::as_type_ptr<ngraph::op::v0::Constant>(input.get_node_shared_ptr());
+        auto tensor = ov::Tensor(input.get_element_type(), input.get_shape());
+        std::copy_n(constant->get_data_ptr<uint8_t>(), constant->get_byte_size(), static_cast<uint8_t*>(tensor.data()));
```

Comment:
Can we avoid an additional copy from constant if use `const_cast` and create tensor on a host memory?

Reply:
I was thinking about it too but on the other hand can we modify the `Tensor` class to be able to create immutable input tensors? This way we could avoid both copy and `const_cast`. I'm not saying to do it in the same PR, perhaps a copy for now (because that's how HostTensors work too) and an additional modification later.

Reply:
At the current moment Tensor has such API, that we always can change the memory.
And from performance perspective will be better to avoid the copy.
Yes I agree that previously `HostTensor` also copied the memory.